pallets/did/src/benchmarking.rs

```diff
@@ -296,11 +295,475 @@ benchmarks! {
 
 		let did_call_signature = ecdsa_sign(AUTHENTICATION_KEY_ID, &did_public_auth_key, did_call_op.encode().as_ref()).expect("Failed to create DID signature from raw ecdsa signature.");
 	}: submit_did_call(RawOrigin::Signed(submitter), Box::new(did_call_op), DidSignature::from(did_call_signature))
+
+	/* set_authentication_key extrinsic */
+	set_ed25519_authentication_key {
+		let block_number = T::BlockNumber::zero();
+		let old_did_public_auth_key = get_ed25519_public_authentication_key();
+		let did_subject: DidIdentifierOf<T> = MultiSigner::from(old_did_public_auth_key).into_account().into();
+
+		// fill up public keys to its max size
+		let mut did_details = generate_base_did_details::<T>(DidVerificationKey::from(old_did_public_auth_key));
+		assert_ok!(did_details.add_key_agreement_keys(get_key_agreement_keys::<T>(T::MaxNewKeyAgreementKeys::get()), block_number));
+		assert_ok!(did_details.update_attestation_key(DidVerificationKey::from(get_ed25519_public_attestation_key()), block_number));
+		assert_ok!(did_details.update_delegation_key(DidVerificationKey::from(get_ed25519_public_delegation_key()), block_number));
+
+		Did::<T>::insert(&did_subject, did_details);
+
+		let new_did_public_auth_key = get_ed25519_public_authentication_key();
```

Comment:
Are we adding the same authentication key that was already set in `did_details`? Shouldn't we try to add a different one for the worst case? I am not sure how much it matters since probably what is more important is what key is used to verify the operation signature.

Reply:
Good catch. In all benchmarks where we wanted to set a new key (authentication, attestation, delegation) I just generated the same key twice 🥶 Should be fixed by e35868ad80fdaf1075baa6328fe2c832fa406f95